lib/workload/stateless/stacks/sample-sheet-check/sample-sheet-check-lambda/src/samplesheet.py

```diff
@@ -341,6 +341,12 @@ def convert_data_to_samples(self):
             index2 = sample_row["index2"] if "index2" in sample_row.keys() else None
             project = sample_row["Sample_Project"] if "Sample_Project" in sample_row.keys() else None
 
+            if not sample_row["Sample_ID"].endswith(sample_row["Sample_Name"]):
```

Comment:
Mainly this, where it checks if the Sample_Name (which should be the library_id) is the same Sample_ID format (which is `{sampleID}_{library_ID}`)